In [1]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_csv("Police Data.csv")

In [6]:
df.head()

stop_date stop_time  country_name driver_gender  driver_age_raw  \
0   1/2/2005      1:55           NaN             M          1985.0   
1  1/18/2005      8:15           NaN             M          1965.0   
2  1/23/2005     23:15           NaN             M          1972.0   
3  2/20/2005     17:15           NaN             M          1986.0   
4  3/14/2005     10:00           NaN             F          1984.0   

   driver_age driver_race     violation_raw violation  search_conducted  \
0        20.0       White          Speeding  Speeding             False   
1        40.0       White          Speeding  Speeding             False   
2        33.0       White          Speeding  Speeding             False   
3        19.0       White  Call for Service     Other             False   
4        21.0       White          Speeding  Speeding             False   

  search_type   stop_outcome is_arrested stop_duration  drugs_related_stop  
0         NaN       Citation       False      0-15 Min               False  
1         NaN       Citation       False      0-15 Min               False  
2         NaN       Citation       False      0-15 Min               False  
3         NaN  Arrest Driver        True     16-30 Min               False  
4         NaN       Citation       False      0-15 Min               False

In [7]:
df.shape

(65535, 15)

# Q) 1. For Speeding , were Men or Women stopped more often?

In [10]:
# Who have been stopped more time men or women for speeding.

df[(df['violation'] == 'Speeding')].driver_gender.value_counts()

M    25517
F    11686
Name: driver_gender, dtype: int64

# Q) 2. Does gender affect who gets searched during a stop?

In [19]:
# First looking for how many times the search was conducted.
df.search_conducted.value_counts()

False    63056
True      2479
Name: search_conducted, dtype: int64

As we can see serch was conducted 2479 times.

In [20]:
# Answer.
df.groupby('driver_gender').search_conducted.sum()

driver_gender
F     366
M    2113
Name: search_conducted, dtype: int64

As we can see here search was conducted 366 times for Females and 2113 for Males.

# Q) 3. Compare the age distribution for each violation.

In [24]:
df.groupby(['violation']).driver_age.describe()

count       mean        std   min   25%   50%   75%  \
violation                                                                    
Equipment             6507.0  31.682957  11.380671  16.0  23.0  28.0  39.0   
Moving violation     11876.0  36.736443  13.258350  15.0  25.0  35.0  47.0   
Other                 3477.0  40.362381  12.754423  16.0  30.0  41.0  50.0   
Registration/plates   2240.0  32.656696  11.150780  16.0  24.0  30.0  40.0   
Seat belt                3.0  30.333333  10.214369  23.0  24.5  26.0  34.0   
Speeding             37120.0  33.262581  12.615781  15.0  23.0  30.0  42.0   

                      max  
violation                  
Equipment            81.0  
Moving violation     86.0  
Other                86.0  
Registration/plates  74.0  
Seat belt            42.0  
Speeding             88.0

As we can see the min, max and the count of age compared with violation.